Makefile

```diff
@@ -125,7 +125,7 @@ pylint:
 
 .PHONY: test
 test:
-	python -m unittest discover -s .
+	pytest tests
```

Comment:
Please add `--doctest-modules` option to enforce doctests.

Reply:
Also please add coverage reports. 
You will need to add the correct plugin (`pytest-cov` ), specify the correct options here (e.g. `--cov` and `--cov-report=`, ...), and place the necessary configs in the repo (e.g. `.coveragerc`). See the aea framework for reference.